### ISSUE TITLE
Formatted printing of byte vectors fails?

### ISSUE THREAD

**Issue by nomennescio** [factor/factor#2267]
Consider this code (having `USE:`d the correct libraries):
`BV{ 1 2 3 } "%[%d, %]" printf`
this will fail with the following error:
`Type check error`
`Object: "0"`
`Object type: string`
`Expected type: fixnum`

whereas
`{ 1 2 3 } "%[%d, %]" printf`
succeeds.

How come? A byte vector is a sequence after all.

**Comment by nomennescio** [comment]
After some testing, this does work:
`BV{ 1 2 3 } { } clone-like "%[%d, %]" printf`

**Comment by erg** [comment]
```
diff --git a/basis/formatting/formatting.factor b/basis/formatting/formatting.factor
index 5b9b722f89..d0673158e3 100644
--- a/basis/formatting/formatting.factor
+++ b/basis/formatting/formatting.factor
@@ -136,7 +136,7 @@ numbers   = sign pad numbers_    => [[ unclip-last prefix compose-all [ fix-sign

 types     = strings|numbers

-lists     = "[%" types ", %]"    => [[ second '[ _ map ", " join "{ " prepend " }" append ] ]]
+lists     = "[%" types ", %]"    => [[ second '[ _ { } map-as ", " join "{ " prepend " }" append ] ]]

 assocs    = "[%" types ": %" types " %]" => [[ [ second ] [ fourth ] bi '[ unzip [ _ map ] dip _ map zip [ ":" join ] map ", " join "{ " prepend " }" append ] ]]

```

**Comment by nomennescio** [comment]
Shouldn't the assocs also not use map-as?

**Comment by mrjbq7** [comment]
Fixed in 552531375721e94d559a54276259fdabf1ef92a3.

**Comment by mrjbq7** [comment]
Also your ``assocs`` comment fixed with 497d6491e6313530c190ef623df9a913f290ff0f.
